### PR TITLE
str: make replace() example more simple

### PR DESCRIPTION
**Pull request by tshepang**

### CODE REVIEW

**Comment by steveklabnik** [comment]
I'm not a fan of losing the flavor, but it is simpler.

**Comment by tshepang** [comment]
thanks @steveklabnik; I found the original hard to read... 'twas too playful :)

File: src/libcollections/str.rs [hunk 1]
```diff
@@ -435,12 +435,9 @@ pub trait StrExt: Index<FullRange, Output = str> {
...
     ///
     /// ```rust
-    /// let s = "Do you know the muffin man,
-    /// The muffin man, the muffin man, ...".to_string();
+    /// let s = "this is old";
     ///
-    /// assert_eq!(s.replace("muffin man", "little lamb"),
-    ///            "Do you know the little lamb,
-    /// The little lamb, the little lamb, ...".to_string());
+    /// assert_eq!(s.replace("old", "new"), "this is new")
```
Comment (barosl) [inline src/libcollections/str.rs]: This caused a test failure. The trailing `;` is missing.

**Comment by barosl** [comment]
This should be re-r+'ed, by the way.

**Pull request merged**